src/algebra/continued_fractions/definitions/basic.lean

```diff
@@ -0,0 +1,321 @@
+/-
+Copyright (c) 2019 Kevin Kappelmann. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Kappelmann
+-/
+import data.seq.seq
+/-!
+# Basic Definitions for Continued Fractions
+
+## Summary
+
+We define generalised, simple, and regular continued fractions and functions to evaluate their
+convergents. We follow the naming conventions from Wikipedia and [wall2018analytic], Chapter 1.
+
+## Main definitions
+
+1. Generalised continued fractions (gcfs)
+2. Simple continued fractions (scfs)
+3. (Regular) continued fractions ((r)cfs)
+4. Computation of convergents using the recurrence relation in `convergents`.
+5. Computation of convergents by directly evaluating the fraction described by the gcf in
+`convergents'`.
+
+## Implementation notes
+
+1. The most commonly used kind of continued fractions in the literature are regular continued
+fractions. We hence just call them `continued_fractions` in the library.
+2. We use sequences from `data.seq` to encode potentially infinite sequences.
+
+## References
+
+- https://en.wikipedia.org/wiki/Generalized_continued_fraction
+- [Wall, H.S., *Analytic Theory of Continued Fractions*][wall2018analytic]
+
+## Tags
+
+numerics, number theory, approximations, fractions
+-/
+
+
+/-
+A *generalised continued fraction* (gcf) is a potentially infinite expression of the form
+
+                                a₀
+                h + --------------------------
+                                  a₁
+                      b₀ + --------------------
+                                    a₂
+                            b₁ + --------------
+                                        a₃
+                                  b₂ + --------
+                                      b₃ + ...
+
+where `h` is called the *head term* or *integer part*, the `aᵢ` are called the
+*partial numerators* and the `bᵢ` the *partial denominators* of the gcf. For convenience, one often
+writes `[h; (a₀, b₀), (a₁, b₁), (a₂, b₂),...]`.
+-/
+
+-- Fix a carrier `α`.
+variable (α : Type*)
+
+/-- We collect a partial numerator `aᵢ` and partial denominator `bᵢ` in a pair `⟨aᵢ,bᵢ⟩`. -/
+protected structure generalized_continued_fraction.pair := (a : α) (b : α)
+
+/- Interlude: define some expected coercions and instances. -/
+namespace generalized_continued_fraction.pair
+open generalized_continued_fraction as gcf
+
+/-- Make a gcf.pair printable. -/
+instance [has_repr α] : has_repr (gcf.pair α) :=
+⟨λ p, "(a : " ++ (repr p.a) ++ ", b : " ++ (repr p.b) ++ ")"⟩
+
+section coe
+/-- Fix another type `β` and assume `α` can be converted to `β`. -/
```

Comment:
I don't think there should be doc string comments above variables. (Not sure how this gets interpreted.)
```suggestion
/- Fix another type `β` and assume `α` can be converted to `β`. -/
```

Reply:
Fixed.